### PR TITLE
[SDK] Update eager flow schema

### PR DESCRIPTION
**Pull request by D-W-**
# Description
This pull request includes changes to the `promptflow` SDK and its associated tests, primarily focusing on modifying the handling of `EagerFlow` and `Flow` in the JSON schemas and the loading functions. The changes remove the `path` property and the `entry` parameter from several methods, and also modify the handling of eager flows in the `FlowSchema` and `EagerFlowSchema` classes. 

Changes to JSON Schemas and Loading Functions:

* `scripts/json_schema/EagerFlow.schema.json` and `scripts/json_schema/Flow.schema.json`: Removed the `path` property from the JSON schemas. [[1]](diffhunk://#diff-aced05d763f0c55b6ac20d919f197673bbdf491f63af1e19dd70c3774661de47L35-L38) [[2]](diffhunk://#diff-aced05d763f0c55b6ac20d919f197673bbdf491f63af1e19dd70c3774661de47L65-R61) [[3]](diffhunk://#diff-6943476b5c0a25caee9c88ff548a284d2e0f78e0874449a57c091a693a21619fL123-L126) [[4]](diffhunk://#diff-6943476b5c0a25caee9c88ff548a284d2e0f78e0874449a57c091a693a21619fL153-R149)
* [`scripts/json_schema/gen_json_schema.py`](diffhunk://#diff-a40f32e9c2dff3899f142d10888a04f41f93d5bab55880380dfe0abf588ea847L148-R148): Modified the `get_required` method to remove the `FlowSchema` from the `cls_list` and to write each schema to a separate file instead of combining them into one schema. [[1]](diffhunk://#diff-a40f32e9c2dff3899f142d10888a04f41f93d5bab55880380dfe0abf588ea847L148-R148) [[2]](diffhunk://#diff-a40f32e9c2dff3899f142d10888a04f41f93d5bab55880380dfe0abf588ea847L157-R164)
* [`src/promptflow/promptflow/_sdk/_load_functions.py`](diffhunk://#diff-823b076b6e4e3c7b30bffee4f0ee2d8e7f32a9fc1d85854c5880aef7681ae2e4L69-L70): Removed the `entry` parameter from the `load_flow` method. [[1]](diffhunk://#diff-823b076b6e4e3c7b30bffee4f0ee2d8e7f32a9fc1d85854c5880aef7681ae2e4L69-L70) [[2]](diffhunk://#diff-823b076b6e4e3c7b30bffee4f0ee2d8e7f32a9fc1d85854c5880aef7681ae2e4L79-R80)
* [`src/promptflow/promptflow/_sdk/entities/_eager_flow.py`](diffhunk://#diff-53d7ff49a8720f52ee715a6f63d5e389750e44c578a907076d435e541f0163eeL50-R54): Modified the `_load` method to remove the `path` and `entry` parameters and to load the `entry` from the `data` parameter instead.
* [`src/promptflow/promptflow/_sdk/entities/_flow.py`](diffhunk://#diff-339752a6e9c04159a62708efed6b385bd91b016a0f0d2edeb32024fd19886a2bL193-R198): Modified the `load` method to remove the `entry` parameter and to raise an exception if a non-YAML file is provided.
* [`src/promptflow/promptflow/_sdk/schemas/_flow.py`](diffhunk://#diff-84af29d72d79c61d55daa91de77a03235646967bcaa7c3299f33adcead97bc53R4-R11): Modified the `FlowSchema` and `EagerFlowSchema` classes to handle eager flows without a `path` property and to validate the `entry` property. [[1]](diffhunk://#diff-84af29d72d79c61d55daa91de77a03235646967bcaa7c3299f33adcead97bc53R4-R11) [[2]](diffhunk://#diff-84af29d72d79c61d55daa91de77a03235646967bcaa7c3299f33adcead97bc53R59-R83)

Changes to Tests:

* `src/promptflow/tests/sdk_cli_test/e2etests/test_flow_run.py` and `src/promptflow/tests/sdk_cli_test/e2etests/test_flow_test.py`: Updated tests to reflect the changes to the handling of eager flows. [[1]](diffhunk://#diff-94a59a05643476869fa3c6bc45466f1582944a935488075e2e63b6a6a196958fL1250-L1252) [[2]](diffhunk://#diff-94a59a05643476869fa3c6bc45466f1582944a935488075e2e63b6a6a196958fL1261) [[3]](diffhunk://#diff-94a59a05643476869fa3c6bc45466f1582944a935488075e2e63b6a6a196958fL1272-L1290) [[4]](diffhunk://#diff-acf3c2285ac2a7051d59e3a69abe02224e2d6d5d69e3a0aabd2afd9311f87657R2) [[5]](diffhunk://#diff-acf3c2285ac2a7051d59e3a69abe02224e2d6d5d69e3a0aabd2afd9311f87657R27-R33) [[6]](diffhunk://#diff-acf3c2285ac2a7051d59e3a69abe02224e2d6d5d69e3a0aabd2afd9311f87657L245-R299)
* `src/promptflow/tests/test_configs/eager_flows/flow_with_additional_includes/flow.dag.yaml`, `src/promptflow/tests/test_configs/eager_flows/flow_with_environment/flow.dag.yaml`, `src/promptflow/tests/test_configs/eager_flows/nested_entry/flow.dag.yaml`, and `src/promptflow/tests/test_configs/eager_flows/nested_entry/my_module/__init__.py`: Updated the flow configuration files to use the new format for the `entry` property. [[1]](diffhunk://#diff-a1826a5844be209b8e39da157975cc6ea52e41463dd40b44a48a63c491e37ebbL1-R1) [[2]](diffhunk://#diff-50cd4769a89c7dc3e460e384b9423e879f9e6bb718c9d3d5e32571e0becfd518L1-R1) [[3]](diffhunk://#diff-728a645780919d47289e213026eaacb1c6079e23568a56532814602a60ef3293R1) [[4]](diffhunk://#diff-7da984bbf7f84ea01603858983d3aee6b4b15a6e21ecda66e158e4aa452a5dc8R1-R3)

Other Changes:

* `src/promptflow/promptflow/_sdk/_pf_client.py` and `src/promptflow/promptflow/_sdk/operations/_flow_operations.py`: Removed the `entry` parameter from several methods. [[1]](diffhunk://#diff-c100045cd304dbaa66dd61b7087f78b813ce01e8566d0b823b293bb35755eccfL122) [[2]](diffhunk://#diff-afdd40a5d0519512dcf9be48bd46c4caaa2291b808687de77896989af63f47e4L81-L82) [[3]](diffhunk://#diff-afdd40a5d0519512dcf9be48bd46c4caaa2291b808687de77896989af63f47e4L92) [[4]](diffhunk://#diff-afdd40a5d0519512dcf9be48bd46c4caaa2291b808687de77896989af63f47e4L158) [[5]](diffhunk://#diff-afdd40a5d0519512dcf9be48bd46c4caaa2291b808687de77896989af63f47e4L175-L189)
Please add an informative description that covers that changes made by the pull request and link all relevant issues.

# All Promptflow Contribution checklist:
- [ ] **The pull request does not introduce [breaking changes].**
- [ ] **CHANGELOG is updated for new features, bug fixes or other significant changes.**
- [ ] **I have read the [contribution guidelines](../CONTRIBUTING.md).**
- [ ] **Create an issue and link to the pull request to get dedicated review from promptflow team. Learn more: [suggested workflow](../CONTRIBUTING.md#suggested-workflow).**

## General Guidelines and Best Practices
- [ ] Title of the pull request is clear and informative.
- [ ] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### Testing Guidelines
- [ ] Pull request includes test coverage for the included changes.

### CODE REVIEW

File: scripts/json_schema/gen_json_schema.py [hunk 1]
```diff
@@ -154,9 +154,11 @@ def get_required(self, obj):
...
         print(target_schema)
-    schema = {
-        "type": "object",
-        "oneOf": schema_list
-    }
-    with open((f"Flow.schema.json"), "w") as f:
-        f.write(json.dumps(schema, indent=4))
+        with open((f"{file_name}.schema.json"), "w") as f:
+            f.write(json.dumps(target_schema, indent=4))
+    # schema = {
```
Comment (wangchao1230) [inline scripts/json_schema/gen_json_schema.py:159]: why this comment here?

Comment (D-W-) [reply scripts/json_schema/gen_json_schema.py:159]: Commented to dump single EagerFlow's schema

**Review by elliotzh — APPROVED** [review]

File: src/promptflow/promptflow/_sdk/schemas/_flow.py [hunk 2]
```diff
@@ -54,10 +56,28 @@ class FlowSchema(BaseFlowSchema):
...
-    # path to flow entry file.
-    path = LocalPathField(required=True)
-    # entry function
+    # entry point, for example: pkg.module:func
     entry = fields.Str(required=True)
+
+    def _deserialize(self, data, **kwargs):
+        data = super()._deserialize(data, **kwargs)
+        if data.get("language", FlowLanguage.Python) == FlowLanguage.Python:
+            PythonEagerFlowEntry().deserialize(data.get("entry"))
```
Comment (elliotzh) [inline src/promptflow/promptflow/_sdk/schemas/_flow.py:82]: looks a little wired to use deserialize here

Comment (elliotzh) [reply src/promptflow/promptflow/_sdk/schemas/_flow.py:82]: We only need to validate 1 field, so maybe directly add validation for all languages in `EagerFlowSchema._validate`

Comment (D-W-) [reply src/promptflow/promptflow/_sdk/schemas/_flow.py]: Changed to `validates_schema` as offline synced.

File: scripts/json_schema/gen_json_schema.py (hunk 1 above)
Comment (D-W-) [reply scripts/json_schema/gen_json_schema.py]: Reverted

**Review by wangchao1230 — APPROVED** [review]

**Review by zhengfeiwang — APPROVED** [review]

**Pull request merged**